tests/platform_tests/test_reboot.py

```diff
@@ -56,9 +56,16 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list, reboot_type=REB
     @param reboot_kwargs: The argument used by reboot_helper
     """
     logging.info("Run %s reboot on DUT" % reboot_type)
+    append_reboot_type_to_q = 1
 
-    reboot(dut, localhost, reboot_type=reboot_type, reboot_helper=reboot_helper, reboot_kwargs=reboot_kwargs)
-    REBOOT_TYPE_HISTOYR_QUEUE.append(reboot_type)
+    try:
+        reboot(dut, localhost, reboot_type=reboot_type, reboot_helper=reboot_helper, reboot_kwargs=reboot_kwargs)
+    except Exception as e:
+        if e is 'DUT {} did not shutdown'.format(dut.hostname):
```

Comment:
Object `e` should be an Exception instance. Is the `is` check robust enough? Probably something like below is more robust:
`'DUT {} did not shutdown'.format(dut.hostname) in repr(e)`